ArduPlane/geofence.cpp

```diff
@@ -280,6 +280,52 @@ bool Plane::geofence_check_maxalt(void)
     return (adjusted_altitude_cm() > (g.fence_maxalt*100.0f) + home.alt);
 }
 
+/*
+  pre-arm check for being inside the fence
+ */
+bool Plane::geofence_prearm_check(void)
+{
+    if (!geofence_enabled()) {
+        gcs().send_text(MAV_SEVERITY_WARNING, "PreArm: Fence not enabled");
+        return false;
+    }
+
+    /* allocate the geo-fence state if need be */
+    if (geofence_state == nullptr || !geofence_state->boundary_uptodate) {
+        geofence_load();
+        if (!geofence_enabled()) {
```

Comment:
null-ptr heck for `geofence_state` here.  I can't think it would ever trigger given the current flow, however.